### PR TITLE
Hmi rota dtu

### PR DESCRIPTION
**Pull request by mjotichmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMIS-880

### Change description ###
Service Now integration 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by subrotomukherjee — APPROVED** [review]

File: infrastructure/logic-app/rota/parameters-dev.json [hunk 1]
```diff
@@ -33,6 +33,21 @@
...
+            "value": "56b756774fbd368057db0b318110c7bd"
+        },
+        "serviceNowUserRoleTypeId": {
+            "value": "c319bc4bdb41834074abffa9bf96199c"
+        },
+        "serviceNowServiceOfferingId": {
+            "value": "138e0c541bc5bc507bdaddf0b24bcb2a"
+        },
+        "serviceNowCallerId": {
+            "value": "1475a0491b743414f0dc85e4464bcb7a"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/rota/parameters-dev.json:50]: Just checking ok with these id's being present?

File: infrastructure/logic-app/rota/parameters-sbox.json [hunk 2]
```diff
@@ -33,6 +33,21 @@
...
+            "value": "56b756774fbd368057db0b318110c7bd"
+        },
+        "serviceNowUserRoleTypeId": {
+            "value": "c319bc4bdb41834074abffa9bf96199c"
+        },
+        "serviceNowServiceOfferingId": {
+            "value": "138e0c541bc5bc507bdaddf0b24bcb2a"
+        },
+        "serviceNowCallerId": {
+            "value": "1475a0491b743414f0dc85e4464bcb7a"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/rota/parameters-sbox.json:50]: Same here

File: infrastructure/logic-app/rota/parameters-test.json [hunk 3]
```diff
@@ -33,6 +33,21 @@
...
+            "value": "56b756774fbd368057db0b318110c7bd"
+        },
+        "serviceNowUserRoleTypeId": {
+            "value": "c319bc4bdb41834074abffa9bf96199c"
+        },
+        "serviceNowServiceOfferingId": {
+            "value": "138e0c541bc5bc507bdaddf0b24bcb2a"
+        },
+        "serviceNowCallerId": {
+            "value": "1475a0491b743414f0dc85e4464bcb7a"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/rota/parameters-test.json:50]: Same here

File: infrastructure/logic-app/sitting-pattern/sitting-loader-param-dev.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        "workflows_hmidtusittingpatterns_name": {
+            "value": "hmi-la-sittingpattern-publisher-dev"
+        },
+        "storageAccountName":{
+            "value": "hmidtudev"
+        },
+        "storageAccessKey":{
+            "reference": {
+                "keyVault": {
+                   "id": "/subscriptions/867a878b-cb68-4de5-9741-361ac9e178b6/resourceGroups/hmi-sharedservices-dev-rg/providers/Microsoft.KeyVault/vaults/hmi-shared-kv-dev"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/sitting-pattern/sitting-loader-param-dev.json:17]: Ok to have id of keyvault?

File: infrastructure/logic-app/sitting-pattern/sitting-loader-param-sbox.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        "workflows_hmidtusittingpatterns_name": {
+            "value": "hmi-la-sittingpattern-publisher-sbox"
+        },
+        "storageAccountName":{
+            "value": "hmidtusbox"
+        },
+        "storageAccessKey":{
+            "reference": {
+                "keyVault": {
+                   "id": "/subscriptions/a8140a9e-f1b0-481f-a4de-09e2ee23f7ab/resourceGroups/hmi-sharedservices-sbox-rg/providers/Microsoft.KeyVault/vaults/hmi-shared-kv-sbox"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/sitting-pattern/sitting-loader-param-sbox.json:17]: Ok to have id of keyvault?

File: infrastructure/logic-app/sitting-pattern/sitting-loader-param-test.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        "workflows_hmidtusittingpatterns_name": {
+            "value": "hmi-la-sittingpattern-publisher-test"
+        },
+        "storageAccountName":{
+            "value": "hmidtutest"
+        },
+        "storageAccessKey":{
+            "reference": {
+                "keyVault": {
+                   "id": "/subscriptions/3eec5bde-7feb-4566-bfb6-805df6e10b90/resourceGroups/hmi-sharedservices-test-rg/providers/Microsoft.KeyVault/vaults/hmi-shared-kv-test"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/sitting-pattern/sitting-loader-param-test.json:17]: Ok to have id of keyvault?

File: infrastructure/logic-app/sitting-pattern/sitting-retriever-param-dev.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        "workflows_hmidtusittingpatterns_name": {
+            "value": "hmi-la-sittingpattern-retriever-dev"
+        },
+        "storageAccountName":{
+            "value": "hmidtudev"
+        },
+        "storageAccessKey":{
+            "reference": {
+                "keyVault": {
+                   "id": "/subscriptions/867a878b-cb68-4de5-9741-361ac9e178b6/resourceGroups/hmi-sharedservices-dev-rg/providers/Microsoft.KeyVault/vaults/hmi-shared-kv-dev"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/sitting-pattern/sitting-retriever-param-dev.json:17]: Ok to have id of keyvault?

File: infrastructure/logic-app/sitting-pattern/sitting-retriever-param-sbox.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        "workflows_hmidtusittingpatterns_name": {
+            "value": "hmi-la-sittingpattern-retriever-sbox"
+        },
+        "storageAccountName":{
+            "value": "hmidtusbox"
+        },
+        "storageAccessKey":{
+            "reference": {
+                "keyVault": {
+                   "id": "/subscriptions/a8140a9e-f1b0-481f-a4de-09e2ee23f7ab/resourceGroups/hmi-sharedservices-sbox-rg/providers/Microsoft.KeyVault/vaults/hmi-shared-kv-sbox"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/sitting-pattern/sitting-retriever-param-sbox.json:17]: Ok to have id of keyvault?

File: infrastructure/logic-app/sitting-pattern/sitting-retriever-param-test.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        "workflows_hmidtusittingpatterns_name": {
+            "value": "hmi-la-sittingpattern-retriever-test"
+        },
+        "storageAccountName":{
+            "value": "hmidtutest"
+        },
+        "storageAccessKey":{
+            "reference": {
+                "keyVault": {
+                   "id": "/subscriptions/3eec5bde-7feb-4566-bfb6-805df6e10b90/resourceGroups/hmi-sharedservices-test-rg/providers/Microsoft.KeyVault/vaults/hmi-shared-kv-test"
```
Comment (Matthew-Summerbell) [inline infrastructure/logic-app/sitting-pattern/sitting-retriever-param-test.json:17]: Ok to have id of keyvault?

File: infrastructure/template/api-op-get-sitting-patterns-policy.xml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+	                        );
+                            JObject recordThree = new JObject(
+	                            new JProperty("id", "dcecc67c-792a-3df8-acd0-8ecbb71f0f6e"),
+	                            new JProperty("oucode", "B01MN00"),
+	                            new JProperty("courtroom_id", "2478"),
+	                            new JProperty("max_slot", "12"),
+	                            new JProperty("max_duration_mins", ""),
+	                            new JProperty("court_session", "TUEAM"),
+	                            new JProperty("rota_business_type_code", "TVLO")
+	                        );
```
Comment (Matthew-Summerbell) [inline infrastructure/template/api-op-get-sitting-patterns-policy.xml:56]: Just wanna check this is test data?

File: infrastructure/logic-app/rota/parameters-dev.json (hunk 1 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/rota/parameters-dev.json:50]: yes - these are env specific for ticket creation

File: infrastructure/logic-app/rota/parameters-sbox.json (hunk 2 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/rota/parameters-sbox.json:50]: see above

File: infrastructure/logic-app/rota/parameters-test.json (hunk 3 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/rota/parameters-test.json:50]: see above

File: infrastructure/logic-app/sitting-pattern/sitting-loader-param-dev.json (hunk 4 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/sitting-pattern/sitting-loader-param-dev.json:17]: yes

File: infrastructure/logic-app/sitting-pattern/sitting-loader-param-sbox.json (hunk 5 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/sitting-pattern/sitting-loader-param-sbox.json:17]: see above

File: infrastructure/logic-app/sitting-pattern/sitting-loader-param-test.json (hunk 6 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/sitting-pattern/sitting-loader-param-test.json:17]: see above

File: infrastructure/logic-app/sitting-pattern/sitting-retriever-param-dev.json (hunk 7 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/sitting-pattern/sitting-retriever-param-dev.json:17]: see above

File: infrastructure/logic-app/sitting-pattern/sitting-retriever-param-sbox.json (hunk 8 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/sitting-pattern/sitting-retriever-param-sbox.json:17]: see above

File: infrastructure/logic-app/sitting-pattern/sitting-retriever-param-test.json (hunk 9 above)
Comment (mjotichmcts) [reply infrastructure/logic-app/sitting-pattern/sitting-retriever-param-test.json:17]: see above

File: infrastructure/template/api-op-get-sitting-patterns-policy.xml (hunk 10 above)
Comment (mjotichmcts) [reply infrastructure/template/api-op-get-sitting-patterns-policy.xml:56]: yes

**Review by Matthew-Summerbell — APPROVED** [review]

**Pull request merged**